src/uint256.rs

```diff
@@ -410,6 +410,10 @@ impl Uint256 {
         let hash_result = keccak256(&bytes1);
         Uint256::from_bytes(&hash_result)
     }
+
+    pub fn non_negative(&self) -> bool {
```

Comment:
I think this will always return true.  (uint256's are non-negative by definition -- hence the name).